### PR TITLE
Enhanced XSS Escaping with HTMLPurifier (and caching)

### PR DESCRIPTION
**Pull request by paragonie-scott**
This pull request does a couple of things:

1. Allows `Nebula_Output::html($string, true)` to process HTML code
   without introducing the risk of XSS. The `purify()` method also works.
2. Implements a key management strategy that won't leak to end users.
   (See `Nebula_Output::getDefaultCacheAdapter()` for more info.) 
3. Caches HTMLPurifier output in the filesystem, in a path based on a
   keyed BLAKE2b hash (modern) or truncated HMAC-SHA512 (legacy systems).
   This directory can safely be deleted to clear the cache.
4. It does all of the above with a `Nebula_Cache` interface, should future
   integration with Redis or Memcached be desired for improved performance.

This will allow Nebula users to quickly and painlessly serve purified HTML
code (which also eliminates the risk of XSS).

For horizontal scaling, you could simply mount an S3 instance at `assets/cache/`
and the key (and caches) would automatically be copied across WordPress
installs. Alternatively, you can duplicate this key across instances as
part of a provisioning process if you wanted predictable cache file paths.

In summary: This improves security without introducing significant overhead
to the performance of Nebula.

(This pull request includes HTMLPurifier v4.10.0 by Edward Z. Yang.)

### CODE REVIEW

**Comment by paragonie-scott** [comment]
This also includes the same commits in #1753. It's only the latter two that are relevant/different.
